### ISSUE TITLE
Connectio lost (HTTP ERROR 599) when downloading large files through proxy

### ISSUE THREAD

**Issue by pisymbol** [jupyterhub/jupyter-server-proxy#281]
### Bug description
Can not download large files through minio behind proxy 

#### Expected behaviour
File should be downloaded

#### Actual behaviour
[E 2021-07-06 10:36:40.919 ServerApp] {
      "Host": "127.0.0.1:8888",
      "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:89.0) Gecko/20100101 Firefox/89.0",
      "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
      "Accept-Language": "en-US,en;q=0.5",
      "Accept-Encoding": "gzip, deflate",
      "Connection": "keep-alive",
      "Referer": "http://127.0.0.1:8888/minio/my-data/",
      "Cookie": "username-127-0-0-1-8888=\"2|1:0|10:1625567544|23:username-127-0-0-1-8888|44:NjE4MDZkY2ExNmQ0NGE1ZjljNzJiMjBjNTMxM2ZkNjc=|9c563251fbaace553f943e1b56085bb0a7f393aca4955c87e3f975004468eb27\"; _xsrf=2|a3b32fd7|d207e1bdf72df12803c565b723922e6d|1625153033",
      "Upgrade-Insecure-Requests": "1"
    }
[E 2021-07-06 10:36:40.919 ServerApp] 599 GET /minio/download/my-data/foo.deb?token=eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJhY2Nlc3NLZXkiOiJtaW5pb2FkbWluIiwiZXhwIjoxNjI1NTY3ODYwLCJzdWIiOiJtaW5pb2FkbWluIn0.OlQeJhkVXCaYs3VLhDe4A00QqrRIzDsR0Lhl3AADbgOtRHyZsEARoyjixU6IDbzsPN0OAuxlxCfT4HA6Ekecvg (172.17.0.1) 4.97ms referer=http://127.0.0.1:8888/minio/my-data/

minio setup:

   'command': _get_minio_command,
        'port': 9000,
        'timeout': 120,
        'new_browser_tab': True,
        'absolute_url': True,
        'launcher_entry': {
            'title': 'MinIO',
            'icon_path': _get_icon_path()
        },

_get_minio_command() just returns 'minio server /data' etc.

I can download small files but not large files. What is the limitation here?

Note that I can download these file using Jupyter "Download" option as well as through the minio CLI tool (mc) with no issue.

The file size is ~130MB.

Note: This notebook is running under docker with 8888 and 9000 exposed etc.

**Comment by manics** [comment]
Hi :wave: This sounds like a configuration problem rather than a bug. Do you mind posting this as a support question on https://discourse.jupyter.org/ instead? Please include as much detail as you can about your entire Jupyter and operating system setup, as the error _may_ result from outside Jupyter.

**Comment by pisymbol** [comment]
I'm a bit confused. Downloads that aren't large work. Downloads that are big don't. This seems like a jupyter proxy issue not a general jupyter issue. Do you still want me to use discourse?

**Comment by manics** [comment]
Yes please, we can re-open this if it's confirmed as a bug in jupyter-server-proxy

**Comment by pisymbol** [comment]
Given it's a 599....but OK! Let me do that.

**Comment by pisymbol** [comment]
https://discourse.jupyter.org/t/can-not-download-large-files-via-jupyter-server-proxy/9874

**Comment by pisymbol** [comment]
I'd like to reopen this. It's a bug/feature requeston on the proxy side:

https://stackoverflow.com/questions/41511974/downloading-large-files-with-tornado-asynchttpclient-streaming-callback-fails

Can we simply add a ```max_body_size``` and ```max_buffer_size``` configurables and then use the above API to set it for the AsyncClient in handlers.py? Pretty please? That will fix it.

Otherwise, the proxy requests are limited to 100MB.

**Comment by manics** [comment]
Thanks for investigating, and your patience!

**Comment by manics** [comment]
I think it's a valid use case that we should support. Is it useful to add a generic dictionary of keyword parameters to support other options in future, or is this the only useful one?

**Comment by ryanlovett** [comment]
Related to #113. Note @yuvipanda 's [comment](https://github.com/jupyterhub/jupyter-server-proxy/pull/113#issuecomment-487356849) about supporting such limits in a generic way without being tied to tornado.

**Comment by pisymbol** [comment]
I completely agree about supporting a dictionary of kwargs here. Tornado's API or whatever API you decide on using to build proxy requests will change over time - there is no reason the proxy should be rigid.

Frankly, if you want to go crazy here you could make the proxy request side more of a driver that can be swapped out by other web frameworks. I'm not necessarily suggesting that but it behooves the "general backend" to be flexible enough to deal with these kinds of tuneables.

So what about a config section that sets a dict of kwargs that gets passed in or configured on each request? Btw, it seems the request and connect timeouts are already somewhat hard-coded - they would be folded into this I believe.

**Comment by pisymbol** [comment]
https://github.com/jupyterhub/jupyter-server-proxy/pull/282

**Comment by pisymbol** [comment]
Hey, could I get some feedback on this issue?

**Comment by unode** [comment]
Sorry for necro-bumping but I just ran into this after making [filebrowser](https://github.com/filebrowser/filebrowser) available as a proxied environment, not only for downloads but uploads as well.

Seeing that some parallel discussions took place in the past, did any configuration or workaround ever get implemented?
